pkg/glob/glob.go

```diff
@@ -228,6 +228,9 @@ segs:
 				if !startingStar.Match(r) {
 					break
 				}
+				if j > len(name) {
```

Comment:
Either this test should be moved above the prior `if !startingStar...` block or the `j := i + len(string(r))` statement should be moved to immediately prior to this line. It is unclear if the calculation of `j` should affect whether or not `startingStar.Match(r)` is called. I don't doubt this change fixes the panic but in its current form it introduces more ambiguity about the logic.